app/src/main/java/com/android/unio/model/user/UserViewModel.kt

```diff
@@ -255,6 +281,18 @@ constructor(
     }
   }
 
+  private fun <T> updateOrAdd(list: MutableList<T>, item: T, operation: (T) -> T) {
+    if (list.contains(item)) {
+      val index = list.indexOf(item)
+      val current = list[index]
+      val newItem = operation(current)
+      list[index] = newItem
+    } else {
+      val newItem = operation(item)
+      list.add(newItem)
+    }
+  }
```

Comment:
Great addition. Add some small documentation to it though, event if it's straightforward to understand what it does.